src/incubator/TransitionAnimator/useAnimatedTranslator.ts

```diff
@@ -0,0 +1,62 @@
+import {useCallback} from 'react';
+import {useSharedValue, useAnimatedStyle, withSpring, withTiming} from 'react-native-reanimated';
+import {Direction} from './useHiddenLocation';
+
+export interface TranslatorProps {
+  initialVisibility: boolean;
+}
+
+const DEFAULT_ANIMATION_VELOCITY = 300;
+const DEFAULT_ANIMATION_CONFIG = {velocity: DEFAULT_ANIMATION_VELOCITY, damping: 18, stiffness: 300, mass: 0.4};
+
+export default function useAnimatedTranslator(props: TranslatorProps) {
+  const {initialVisibility} = props;
+
+  // Has to start at {0, 0} with {opacity: 0} so layout can be measured
+  const translateX = useSharedValue<number>(0);
+  const translateY = useSharedValue<number>(0);
+
+  const visible = useSharedValue<boolean>(initialVisibility);
+  
+  const init = useCallback((to: {x: number; y: number},
+    callback: (isFinished: boolean) => void,
+    animationDirection?: Direction) => {
+    'worklet';
+    if (animationDirection) {
+      if (['left', 'right'].includes(animationDirection)) {
+        translateX.value = withTiming(to.x, {duration: 0}, callback);
+      } else if (['top', 'bottom'].includes(animationDirection)) {
+        translateY.value = withTiming(to.y, {duration: 0}, callback);
+      }
+    }
+
+    visible.value = true;
+  },
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  []);
+
+  const animate = useCallback((to: {x: number; y: number},
```

Comment:
The arguments order here feels a little weird. 
You have `to` the end callback and then the direction.. 
Don't you think it's more appropriate to have (to, direction, endCallback) ?

Reply:
Yes, but the direction can be undefined and `A required parameter cannot follow an optional parameter.`

Reply:
Why is `direction` argument undefined? 
In `animate` implementation, the method does nothing if direction was not passed...
Also, in all usages, you always pass the direction.

Reply:
I guess you're right, fixed.

Reply:
@ethanshar 
I thought I fixed this, I see in master it's not fixed, I'll fix in a future PR.